lib/galaxy_test/base/populators.py

```diff
@@ -1440,7 +1444,7 @@ def create_workflow_response(self, workflow: Dict[str, Any], **create_kwds) -> R
         upload_response = self._post("workflows/upload", data=data)
         return upload_response
 
-    def upload_yaml_workflow(self, has_yaml, **kwds) -> str:
+    def upload_yaml_workflow(self, has_yaml: HasYamlT, **kwds) -> str:
```

Comment:
The `has_*` variable name suggests a boolean type. I know this is old code, but since you're improving it, do you think it would make sense to rename it? (now that there're type hints, the name mismatch stands out, I think). We call that method from multiple places, but I don't think we use the `has_yaml` to pass a keyword argument (including gxformat2).

Reply:
`yaml_content`? To interpret "has_yaml" correctly, I feel I have to read "has *this* yaml" or "has yaml and *here it is*"; when I read "has_yaml", I see a flag argument whose value alters how the function behaves. But if you feel the code reads better the way it is, let's keep it.